frontend/src/components/PageHeader/PageHeader.tsx

```diff
@@ -0,0 +1,134 @@
+
+export interface IPageHeaderProps {
+  headerTitle: string;
+  children: React.ReactElement | React.ReactElement[];
+}
+
+export default function PageHeader(props: IPageHeaderProps) {
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const navigate = useNavigate();
+  const { isLoggedIn, logIn, logOut } = useContext(UserLoginContext);
+
+  const handleMenu = (event: React.MouseEvent<HTMLElement>) => {
+    console.log(event);
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleMyAccount = () => {
+    navigate("/seller/1");
+    handleClose();
+  };
+
+  const handleHome = () => {
+    navigate("/");
+    handleClose();
+  };
+
+  const handleLogin = (e: React.MouseEvent<HTMLElement>) => {
+    e.preventDefault();
+    logIn("Token");
+    return false;
+  };
+
+  const handleLogout = (e: React.MouseEvent<HTMLElement>) => {
+    e.preventDefault();
+    logOut();
+    handleClose();
+  };
+
+  return (
+    <React.Fragment>
+      <AppBar>
+        <Container>
+          <Toolbar>
+            <IconButton
+              size="large"
+              edge="start"
+              color="inherit"
+              aria-label="menu"
+              onClick={handleHome}
+            >
+              <StorefrontIcon
+                fontSize="large"
+                sx={{ display: { md: "flex" } }}
+              />
+            </IconButton>
+            <Typography variant="h5" component="div" sx={{ flexGrow: 1 }}>
+              {props.headerTitle}
+            </Typography>
+            {/* TODO - Make this Log in button go to a page*/}
+            {!isLoggedIn && (
+              <button
+                className="button"
+                onClick={handleLogin}
+                data-testid="login button"
+              >
+                Log In
+              </button>
+            )}
+            <IconButton
+              size="large"
+              aria-label="account of current user"
+              aria-controls="menu-appbar"
+              aria-haspopup="true"
+              onClick={handleMenu}
+              color="inherit"
+            >
+              <AccountCircle />
+            </IconButton>
+            <Menu
+              id="menu-appbar"
+              anchorEl={anchorEl}
+              anchorOrigin={{
+                vertical: "top",
+                horizontal: "right",
+              }}
+              keepMounted
+              transformOrigin={{
+                vertical: "top",
+                horizontal: "right",
+              }}
+              open={Boolean(anchorEl)}
+              onClose={handleClose}
+            >
+              {isLoggedIn && (
+                <MenuItem onClick={handleMyAccount}>My Account</MenuItem>
+              )}
+              {isLoggedIn && (
+                <MenuItem onClick={handleLogout}>Log Out</MenuItem>
+              )}
+            </Menu>
```

Comment:
![image](https://user-images.githubusercontent.com/72239525/223462612-5b2d236b-b9e6-421b-b975-1a7c0e36a59a.png)

When not logged in, clicking on the icon shows this. Could you change it so that the icon-button that opens the menu is hidden when not logged in?

Reply:
![chrome-capture-2023-2-7](https://user-images.githubusercontent.com/39808977/223532626-26e6d201-485e-4346-adaf-01c130241343.gif)

I am have updated

Reply:
Good work!